finality-aleph/src/data_io/data_store.rs

```diff
@@ -309,6 +309,27 @@ where
                             debug!(target: "aleph-data-store", "Requesting a stale block {:?} after it has been missing for {:?} secs.", block, time_waiting.as_secs());
                             self.block_requester
                                 .request_stale_block(block.hash, block.num);
+                        } else {
+                            // The whole branch has been imported, what's holding us must be that the parent of the base is not finalized
+                            let bottom_block = proposal.bottom_block();
+                            if let Ok(parent_hash) =
+                                self.chain_info_provider.get_parent_hash(&bottom_block)
+                            {
+                                let parent_num = bottom_block.num - NumberFor::<B>::one();
+                                if self
+                                    .chain_info_provider
+                                    .get_finalized_at(parent_num)
+                                    .is_err()
```

Comment:
this could be more concise if we would add a method `is_finalized(hash) -> bool` to the chain_info_provider.

Reply:
That's correct, but I actually changed this a little to log more accurately, and now `bool` would not be enough. So not adding it :)